File: 2026/04-April/22-Wednesday/solution.java
// Words Within Two Edits of Dictionary

import java.util.*;

class Solution {
    public List<String> twoEditWords(String[] queries, String[] dictionary) {
        List<String> res = new LinkedList<>();
        
        for (int i = 0; i < queries.length; i++) {
            String q = queries[i];
            
            for (String d : dictionary) {
                if (check(q, d)) {
                    res.add(q);
                    break; 
                }
            }
        }
        
        return res;
    }
    
    private boolean check(String a, String b) {
        int changes = 0;
        
        for (int i = 0; i < a.length(); i++) {
            if (a.charAt(i) != b.charAt(i)) {
                changes++;
            }
            if (changes > 2) return false;
        }
        
        return true;
    }
}

File: 2026/04-April/22-Wednesday/solution.py
# Words Within Two Edits of Dictionary

class Solution:
    def twoEditWords(self, queries, dictionary):
        
        answer = []
        
        for word in queries:
            for target in dictionary:
                mismatch = 0 
                
                for a, b in zip(word, target):
                    if a != b:
                        mismatch += 1
                        if mismatch > 2:
                            break
                
                if mismatch <= 2:
                    answer.append(word)
                    break
        
        return answer
